TinyInsights/ApplicationInsightsProvider.cs

```diff
@@ -498,7 +513,29 @@ public async Task TrackEventAsync(string eventName, Dictionary<string, string>?
         }
     }
 
-    public async Task TrackPageViewAsync(string viewName, Dictionary<string, string>? properties = null, TimeSpan? duration = null)
+    public async Task TrackPageVisitTime(string pageFullName, string pageDisplayName, double pageVisitTime)
+    {
+        if (Client is null)
+        {
+            return;
+        }
+
+        var properties = new Dictionary<string, string>
+        {
+            { "Page", pageFullName },
+            { "DisplayName", pageDisplayName },
+        };
+
+        var metrics = new Dictionary<string, double>
+        {
+            { "Duration", pageVisitTime }
+        };
+
+        Client.TrackEvent("PageVisitTime", properties, metrics);
```

Comment:
@dhindrik I've mostly copied what MS has done in their react library, the main difference is they are using `TrackMetric` whereas I've used `TrackEvent`, with custom metric parameters.

`TrackMetric` can be used instead just by doing this - 
```csharp
  Client.TrackMetric("PageVisitTime", pageVisitTime, properties);
```

I decided not to do this, due to the comment on the `TrackMetric` method. As I haven't used metrics before I thought it be safer to stick with events, but feel free to change it to `TrackMetric` instead _(I'm just not sure about the ramifications of it)_

![image](https://github.com/user-attachments/assets/21a12842-52b4-4a45-ad0e-e03a04991f87)